cmd/main.go

```diff
@@ -116,6 +143,10 @@ func main() {
 		c.NextProtos = []string{"http/1.1"}
 	}
 
+	if !flagsSet["metrics-bind-address"] {
```

Comment:
I think we can move this logic to the `apply` function as well, right, to make that function more pure. But it's ok to defer this with the deprecation PR together, this one is good enough to me. Feel free to unhold the PR.
/lgtm
/approve
/hold

Thanks @ardaguclu also cc @ahg-g especially about the deprecation decision.